internal/infrastructure/kubernetes/proxy_service.go

```diff
@@ -46,11 +46,13 @@ func (i *Infra) expectedProxyService(infra *ir.Infra) (*corev1.Service, error) {
 	// Get annotations
 	var annotations map[string]string
 	provider := infra.GetProxyInfra().GetProxyConfig().GetProvider()
-	svcCfg := provider.GetKubeResourceProvider().EnvoyService
-	if svcCfg != nil {
-		annotations = svcCfg.Annotations
-
+	envoyServiceConfig := provider.GetKubeResourceProvider().EnvoyService
+	if envoyServiceConfig.Annotations != nil {
+		annotations = envoyServiceConfig.Annotations
 	}
+	serviceSpec := serviceSpecByServiceType(envoyServiceConfig.Type)
+	serviceSpec.Ports = ports
```

Comment:
move to `serviceSpecByServiceType`

Reply:
i think its unnecessary,  `serviceSpecByServiceType` func just do different by type

Reply:
+1 either move all service spec code to one func (`expectedProxyServiceSpec()`) or leave them here (and rm `serviceSpecXX` ), else it makes it very hard to understand final intent